### PR TITLE
Testing the IR type-checker's handling of variants and case constructs

### PR DESCRIPTION
**Pull request by frank-emrich**
This mostly contains tests for the IR type-checker for variants and case constructs.

In the process, I discovered the following bugs:
- In `TypeUtils.split_row`, removing a field from an open row would not make it `Absent`, but just remove it altogether. Hence, the resulting row may become ill-kinded (if we had a mechanism for checking that, which we don't)
- The IR type-checker didn't allow matching over a variant type with presence polymorphism in any of its fields. However, the frontend does allow this, if the presence polymorphism only affects labels handled by the default case. I've added test cases for both type-checkers.
- The IR type-checker didn't check that the binder for the default case (if it exists) has the right type, which can be deduced from the type of the expression matched over and the variants present in the match clauses.

I also realized that all the front-end tests for patterns pass the IR type-checker now, which is why I've enabled IR type-checking for these tests.

### CODE REVIEW

**Comment by jamescheney** [comment]
As discussed in Monday's meeting, this is still planned to be merged once updating work is done.

**Comment by frank-emrich** [comment]
This is currently blocked by #955, will merge after that is fixed.

File: tests/unit/unitTests.ml
Comment (dhil) [inline tests/unit/unitTests.ml:30]: Can we split this unit test definition into two, such that the database and non-database suites are in separate files?

**Review by dhil — CHANGES_REQUESTED** [review]

File: tests/unit/unitTests.ml
Comment (frank-emrich) [reply tests/unit/unitTests.ml:30]: Not sure if I understand what you mean. None of these tests should interact with a database?

Comment (dhil) [reply tests/unit/unitTests.ml:30]: So the lens tests do not use the database?

File: .github/workflows/default.yml
Comment (dhil) [inline .github/workflows/default.yml]: I think YAML insists that you line up L107-109 on the same column.

Comment (dhil) [reply .github/workflows/default.yml]: You also need to install `ounit2` in the job, e.g. on line https://github.com/links-lang/links/blob/3ba9c18bd77dc6ec7aa1d655d5ef404308c07948/.github/workflows/default.yml#L91

File: tests/unit/unitTests.ml
Comment (frank-emrich) [reply tests/unit/unitTests.ml]: as discussed elsewhere: you're right, and I split the unit tests into those using the database and those that don't

**Review by dhil — APPROVED** [review]

**Pull request merged**
